File: src/SOP8/Base/Account/Base.Account.IBLL/IProcessManager.cs
using Base.Model.Account;
using Response;
using System.Collections.Generic;

namespace Base.Account.IBLL
{
    using Request;
    using Response;
    using Interface;
    using Models;

    public interface IProcessManager
    {
        ResponseLoginKey RequestLoginKey(RequestLoginKey data, bool isExternalLogin);
        string GetLoginKey(long num);
        LoginResult Login(RequestLogin data, string strExternalLoginUrl, bool autoLogin);
        LoginResult CheckLoginSession(int nUserNo, string strSessionKey);
        ResponseOption GetOption(RequestOption data);
        MessageResult SaveOptions(SaveOption data);
        ResponseSite GetAllSites();
        ResponseUserList GetUserList(RequestUserList data);
        ResponseGradeList GetGradeList();
        MessageResult UpdateUserInfo(RequestUpdateUserInfo data);
        MessageResult DeleteUser(RequestDeleteUser data);
        ResponseAccountUser AddNewUser(RequestNewUser data, IUserCreator userCreator, IPasswordPolicy passwordPolicy);
        MessageResult ChangePassword(RequestChangePassword data);
        string MakeRandomPassword(IPasswordPolicy passwordPolicy, int userNo, string strID, ref string strSalt, out string strErrorMessage);
        MessageResult MakeTemporaryPasswordWithSMS(IUserCreator userCreator, IPasswordPolicy passwordPolicy, string strUserName, string strPhoneNumber);
        MessageResult MakeTemporaryPasswordWithEmail(IUserCreator userCreator, IPasswordPolicy passwordPolicy, string strUserName, string strEmail);
    }
}

File: src/SOP8/광양/EgovLogin/base-account-ibll/src/main/java/egovframework/base/account/ibll/IProcessManager.java
package egovframework.base.account.ibll;

import egovframework.base.model.account.*;
import egovframework.base.account.ibll.request.*;
import egovframework.base.account.ibll.response.*;
import egovframework.base.account.ibll.interface_.*;
import egovframework.base.account.ibll.models.*;
import egovframework.base.response.*;
import egovframework.base.generic.Holder;

public interface IProcessManager {

    ResponseLoginKey requestLoginKey(RequestLoginKey data, boolean isExternalLogin);

    String getLoginKey(long num);

    LoginResult login(RequestLogin data, String strExternalLoginUrl, boolean autoLogin);

    LoginResult checkLoginSession(int nUserNo, String strSessionKey);

    ResponseOption getOption(RequestOption data);

    MessageResult saveOptions(SaveOption data);

    ResponseSite getAllSites();

    ResponseUserList getUserList(RequestUserList data);

    ResponseGradeList getGradeList();

    MessageResult updateUserInfo(RequestUpdateUserInfo data);

    MessageResult deleteUser(RequestDeleteUser data);

    ResponseAccountUser addNewUser(RequestNewUser data, IUserCreator userCreator, IPasswordPolicy passwordPolicy);

    MessageResult changePassword(RequestChangePassword data);

    String makeRandomPassword(IPasswordPolicy passwordPolicy, int userNo, String strID, StringBuilder strSalt, Holder<String> errorMessage);

    MessageResult makeTemporaryPasswordWithSMS(IUserCreator userCreator, IPasswordPolicy passwordPolicy, String strUserName, String strPhoneNumber);

    MessageResult makeTemporaryPasswordWithEmail(IUserCreator userCreator, IPasswordPolicy passwordPolicy, String strUserName, String strEmail);
}